dattri/algorithm/data_shapley.py

```diff
@@ -0,0 +1,135 @@
+"""This module implement data shapley valuation."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import List
+
+import torch
+from tqdm import tqdm
+
+from .base import BaseAttributor
+from .utils import _check_shuffle
+
+
+class KNNShalpeyAttributerExact(BaseAttributor):
+    """KNN Data Shapley Attributer."""
+
+    def __init__(
+        self,
+        k: int,
+        distance_func: Callable,
+    ) -> None:
+        """Initialize the AttributionTask.
+
+        KNN Data Shapley Valuation is task-agnostic.
+
+        Args:
+            k (int): The number of neighbors in KNN model.
+            distance_func (Callable): Customizable function used for
+                distance calculation in KNN. The function can be quite
+                flexible in terms of what is calculated, but it should
+                take two batches of data as input.
+                A typical example is as follows:
+                ```python
+                def f(train_batch, test_batch):
+                    coord1 = train_batch[0]
+                    coord2 = test_batch[0]
+                    return torch.cdist(coord1, coord2)
+                ```.
+        """
+        self.k = k
+        self.distance_func = distance_func
+
+    def cache(self) -> None:
+        """Precompute and cache some values for efficiency."""
+
+    def attribute(
+        self,
+        train_dataloader: torch.utils.data.DataLoader,
+        test_dataloader: torch.utils.data.DataLoader,
+        train_labels: List[int],
+        test_labels: List[int],
```

Comment:
labels should already be included in the dataloaders? why need additional input here